src/preact/hooks/useStateSafe.js

```diff
@@ -0,0 +1,33 @@
+import {useCallback, useEffect, useRef, useState} from '#preact';
+
+/**
+ * Same as useState, but ignores setState once the component is unmounted.
+ *
+ * This avoids React's "called setState() on an unmounted component" console error
+ * @param {S|function():S} initial
+ * @return {{0: S, 1: function((S|function(S):S)):undefined}}
+ * @template S
+ */
+export function useStateSafe(initial) {
+  const isMounted = useRef(false);
+  useEffect(() => {
+    isMounted.current = true;
+
+    return () => {
+      isMounted.current = false;
```

Comment:
React logs an error any time `setState` is called while unmounted, and so the purpose of this hook is to silence this warning (and ignore the state) when dealing with async logic.

I have no experience with react reparenting ... after researching this a little bit, it doesn't appear to be something built-in to React, and might only be possible with unsafe 3rd party libraries.  I haven't seen any examples of such behavior in this project.
If reparenting does temporarily unmount the component, ideally it would be synchronously remounted, not allowing for such a race condition.  
What do you think?  Should this be safe to leave-in?

Reply:
Ha, I'm getting my concepts mixed up with regular elements. You're right that reparenting will just create a new component, so we have no issues. LGTM.